AffiliateWP/affiliatewp-allowed-products
# Adds a check to confirm referral products is an array before iterating. #40

Issue #40


**Housekeeping**


**Proposed changes**



## To Test

1. Clone and activate this branch
1. Run the filter that actually calls this function, using referrals with different 
1. Confirm that the function call does not cause any uncaught errors (see snippet below)
1. Create a new referral using an allowed product and confirm the message still sends as-expected

```php

use AffWP\Referral;

$test = [
	// Test against non-array
	(object) [ 'products' => 'invalid product type' ],
	// Test against empty
	(object) [],
	(object) [ 'products' => [ [ 'array' => 'without id' ] ] ],
	(object) [ 'products' => (object) [ 'object' => 'without id' ] ],
	(object) [ 'products' => [ [ 'this_is_valid' => 'because it has an id', 'id' => 100 ] ] ],
	(object) [ 'products' => [ (object) [ 'this_is_valid' => 'because it has an id', 'id' => 100 ] ] ],
];


add_filter( 'the_content', function () use ( $test ) {
	foreach ( $test as $referral ) {
		var_dump( apply_filters( 'affwp_notify_on_new_referral', false, new Referral( $referral ) ) );
	}
} );
```
